tests/pytests/unit/utils/test_gitfs_locks.py

```diff
@@ -0,0 +1,613 @@
+"""
+These only test the provider selection and verification logic, they do not init
+any remotes.
+"""
+
+import logging
+import os
+import pathlib
+import signal
+import time
+
+import pytest
+from saltfactories.utils import random_string
+
+import salt.ext.tornado.ioloop
+import salt.fileserver.gitfs
+import salt.utils.files
+import salt.utils.gitfs
+import salt.utils.path
+import salt.utils.platform
+import salt.utils.process
+from salt.utils.immutabletypes import freeze
+from salt.utils.verify import verify_env
+
+try:
+    import pwd
+except ImportError:
+    import salt.utils.win_functions
+
+pytestmark = [pytest.mark.skip_unless_on_linux]
```

Comment:
Why are we only testing the locks on linux?

Reply:
because ported this from old unittest and added the SIGTERM tests which isn't really applicable on Windows

Reply:
> because ported this from old unittest and added the SIGTERM tests which isn't really applicable on Windows

But they are applicable on MacOS